### PR TITLE
Task2 Webpack lecture

### PR DESCRIPTION
**Pull request by liaukou**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +20,24 @@
   },
   "homepage": "https://github.com/liaukou/ReactMentoring#readme",
   "dependencies": {
-    "express": "^4.16.3"
+    "express": "^4.16.3",
+    "html-webpack-plugin": "^3.2.0",
```
Comment (spzm) [inline package.json]: to dev dependencies

File: package.json [hunk 2]
```diff
@@ -18,6 +20,24 @@
   },
   "homepage": "https://github.com/liaukou/ReactMentoring#readme",
   "dependencies": {
-    "express": "^4.16.3"
+    "express": "^4.16.3",
```
Comment (spzm) [inline package.json]: to dev dependencies

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        entry: "./",
+        mode: isProduction ? "production" : "development",
+        devtool: isProduction ? "none" : "source-map",
+
+        resolve: {
+            extensions: [".js"]
+        },
+
+        module: {
+            rules: [
```
Comment (spzm) [inline webpack.config.js:21]: I suggest to setup split chunk to have two bundles: `vendor.js` with libs and `main.js` with application code.

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                {
+                    test: /\.(js|jsx)$/,
+                    loader: "babel-loader",
+                    exclude: /node_modules/
+                }
+            ]
+        },
+
+        plugins: [
```
Comment (spzm) [inline webpack.config.js]: please add minification step for production

File: webpack.config.js (hunk 3 above)
Comment (spzm) [reply webpack.config.js:21]: also would be nice to have hash in bundle name - good point for cache optimizations in the future

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            new HtmlWebpackPlugin({
+                title: "App",
```
Comment (spzm) [inline webpack.config.js]: Please don't hardcode title name. Extract it to a variable or better to environment variable

File: index.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
```
Comment (spzm) [inline index.html]: I suggest to store `index.html` in src folder

File: package.json [hunk 7]
```diff
@@ -4,8 +4,10 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "start": "node index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "build": "npm run clean && webpack --env.prod",
+    "clean": "rm -rf ./dist",
+    "start": "webpack-dev-server",
+    "test": "jest"
```
Comment (spzm) [inline package.json:10]: Please setup jest configuration in separate file. Don't trust deаults

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                hash: true,
+                template: path.resolve(__dirname, "./index.html")
+            })
+        ],
+
```
Comment (spzm) [inline webpack.config.js]: good to try custom minification setup through: `optimization.mininize`

File: webpack.config.js (hunk 3 above)
Comment (liaukou) [reply webpack.config.js:21]: Thanks, done.

File: package.json (hunk 7 above)
Comment (liaukou) [reply package.json:10]: Thanks, done.

**Pull request merged**